etna/datasets/tsdataset.py

```diff
@@ -398,6 +398,10 @@ def inverse_transform(self):
             for transform in reversed(self.transforms):
                 tslogger.log(f"Inverse transform {repr(transform)} is applied to dataset")
                 self.df = transform.inverse_transform(self.df)
+                columns_after = set(self.columns.get_level_values("feature"))
+                self._regressors = list(set(self._regressors).union(columns_after))
```

Comment:
In fact not all the columns are regressors, so it is not the correct way to fix it. However I have no guess how to fix it correctly)